pkg/sql/sqlstats/insights/integration/integration_test: TestTransactionInsightsFailOnCommit failed

pkg/sql/sqlstats/insights/integration/integration_test.TestTransactionInsightsFailOnCommit [failed](https://tanzanite.cluster.engflow.com/invocations/default/05b0e99b-d107-4e85-a3f1-e9a9d88b8559?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NxbHN0YXRzL2luc2lnaHRzL2ludGVncmF0aW9uOmludGVncmF0aW9uX3Rlc3Q=) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
=== RUN   TestTransactionInsightsFailOnCommit
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTransactionInsightsFailOnCommit1938775811
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    insights_test.go:563: -- test log scope end --
test logs left over in: outputs.zip/logTestTransactionInsightsFailOnCommit1938775811
--- FAIL: TestTransactionInsightsFailOnCommit (23.07s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true
    --- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true (0.16s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true/40001_error_exists_in_txn_insights
    insights_test.go:499: 
        	Error Trace:	pkg/sql/sqlstats/insights/integration/integration_test/pkg/sql/sqlstats/insights/integration/insights_test.go:499
        	Error:      	Not equal: 
        	            	expected: "SELECT * FROM myusers WHERE city = '_' ; UPDATE myusers SET name = '_' WHERE city = '_'"
        	            	actual  : "UPDATE myusers SET name = '_' WHERE city = '_'"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-SELECT * FROM myusers WHERE city = '_' ; UPDATE myusers SET name = '_' WHERE city = '_'
        	            	+UPDATE myusers SET name = '_' WHERE city = '_'
        	Test:       	TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true/40001_error_exists_in_txn_insights
        --- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true/40001_error_exists_in_txn_insights (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransactionInsightsFailOnCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34947

pkg/sql/sqlstats/insights/integration/integration_test.TestTransactionInsightsFailOnCommit [failed](https://tanzanite.cluster.engflow.com/invocations/default/613e74e8-563c-43e0-b135-d85141636ac3?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NxbHN0YXRzL2luc2lnaHRzL2ludGVncmF0aW9uOmludGVncmF0aW9uX3Rlc3Q=) on master @ [3060756a084cc437114d64265e79ace5720a7317](https://github.com/cockroachdb/cockroach/commits/3060756a084cc437114d64265e79ace5720a7317):


```
=== RUN   TestTransactionInsightsFailOnCommit
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTransactionInsightsFailOnCommit2310668375
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    insights_test.go:562: -- test log scope end --
test logs left over in: outputs.zip/logTestTransactionInsightsFailOnCommit2310668375
--- FAIL: TestTransactionInsightsFailOnCommit (13.81s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false
    --- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false (3.30s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false/40001_error_exists_in_txn_insights
    insights_test.go:498: 
        	Error Trace:	pkg/sql/sqlstats/insights/integration/integration_test/pkg/sql/sqlstats/insights/integration/insights_test.go:498
        	Error:      	Not equal: 
        	            	expected: "SELECT * FROM myusers WHERE city = '_' ; UPDATE myusers SET name = '_' WHERE city = '_'"
        	            	actual  : "UPDATE myusers SET name = '_' WHERE city = '_'"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-SELECT * FROM myusers WHERE city = '_' ; UPDATE myusers SET name = '_' WHERE city = '_'
        	            	+UPDATE myusers SET name = '_' WHERE city = '_'
        	Test:       	TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false/40001_error_exists_in_txn_insights
        --- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false/40001_error_exists_in_txn_insights (0.14s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true
    --- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true (0.21s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true/40001_error_exists_in_txn_insights
    insights_test.go:498: 
        	Error Trace:	pkg/sql/sqlstats/insights/integration/integration_test/pkg/sql/sqlstats/insights/integration/insights_test.go:498
        	Error:      	Not equal: 
        	            	expected: "SELECT * FROM myusers WHERE city = '_' ; UPDATE myusers SET name = '_' WHERE city = '_'"
        	            	actual  : "UPDATE myusers SET name = '_' WHERE city = '_'"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-SELECT * FROM myusers WHERE city = '_' ; UPDATE myusers SET name = '_' WHERE city = '_'
        	            	+UPDATE myusers SET name = '_' WHERE city = '_'
        	Test:       	TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true/40001_error_exists_in_txn_insights
        --- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=true/40001_error_exists_in_txn_insights (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransactionInsightsFailOnCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -490,12 +490,16 @@ SELECT query,
        COALESCE(last_error_code, '') last_error_code,
        COALESCE(last_error_redactable, '') last_error
 FROM crdb_internal.node_txn_execution_insights 
-WHERE app_name = $1`, appName)
+WHERE app_name = $1
+AND query = 'SELECT * FROM myusers WHERE city = _ ; UPDATE myusers SET name = _ WHERE city = _'`, appName)
 
 					return row.Scan(&query, &problems, &status, &errorCode, &errorMsg)
 				})
 
-				require.Equal(t, "SELECT * FROM myusers WHERE city = _ ; UPDATE myusers SET name = _ WHERE city = _", query)
+				require.Equalf(t,
+					"SELECT * FROM myusers WHERE city = _ ; UPDATE myusers SET name = _ WHERE city = _", query,
+					"unexpected txn insight found - query: %s, problems: %s, status: %s, errCode: %s, errMsg: %s",
+					query, problems, status, errorCode, errorMsg)
 				expectedProblem := "{FailedExecution}"
 				replacedSlowProblems := problems
 				if problems != expectedProblem {

